src/util/terminationPreprocessing.ml

```diff
@@ -0,0 +1,65 @@
+open GoblintCil
+
+module VarToStmt = Map.Make(CilType.Varinfo) (* maps varinfos (= loop counter variable) to the statement of the corresponding loop*)
+
+let extract_file_name s =                    (*There still may be a need to filter more chars*)
+  let ls = String.split_on_char '/' s in    (*Assuming '/' as path seperator*)
+  let ls = List.rev ls in
+  let s' = List.nth ls 0 in
+  let ls = String.split_on_char '.' s' in
+  let s' = List.nth ls 0 in
+  let without_spaces = String.split_on_char ' ' s' in
+  let s' = String.concat "" without_spaces in
+  s'
+
+let show_location_id l =
+  string_of_int l.line ^ "_" ^ string_of_int l.column ^ "-file" ^ "_" ^  extract_file_name l.file
+
+class loopCounterVisitor lc (fd : fundec) = object(self)
+  inherit nopCilVisitor
+
+  method! vstmt s =
+
+    let specialFunction name =
+      { svar  = makeGlobalVar name (TFun(voidType, Some [("exp", intType, [])], false,[]));
+        smaxid = 0;
+        slocals = [];
+        sformals = [];
+        sbody = mkBlock [];
+        smaxstmtid = None;
+        sallstmts = [];
+      } in
+
+    let min_int_exp = Const(CInt(Cilint.shift_left_cilint Cilint.mone_cilint ((bytesSizeOfInt IInt)*8-1), IInt, None)) in
+
+    let f_bounded  = Lval (var (specialFunction "__goblint_bounded").svar) in
+
+    let action s = match s.skind with
+      | Loop (b, loc, eloc, _, _) ->
+        let name = "term"^show_location_id loc in
+        let typ = Cil.intType in
+        let v = (Cil.makeLocalVar fd name typ) in (*Not tested for incremental mode*)
+        let init_stmt = mkStmtOneInstr @@ Set (var v, min_int_exp, loc, eloc) in
+        let inc_stmt = mkStmtOneInstr @@ Set (var v, increm (Lval (var v)) 1, loc, eloc) in
+        let inc_stmt2 = mkStmtOneInstr @@ Set (var v, increm (Lval (var v)) 1, loc, eloc) in
+        let exit_stmt = mkStmtOneInstr @@ Call (None, f_bounded, [Lval (var v)], loc, locUnknown) in
+        (match b.bstmts with
+         | s :: ss ->   (*duplicate increment statement here to fix inconsistencies in nested loops*)
+           b.bstmts <- exit_stmt :: inc_stmt :: s :: inc_stmt2 :: ss;
+         | ss ->
+           b.bstmts <- exit_stmt :: inc_stmt :: ss;
```

Comment:
What inconsistencies? This seems odd.

Reply:
I don't remember the details. Probably, @Nimmseasy knows best about it.

Reply:
I changed the code to only introduce one increment statement, as I could not think of a case where the additional would be necessary. There was also no test case that became unsound due to this change. I also started a run on sv-comp.